xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -4640,6 +4655,10 @@ std::string CDVDPlayer::GetPlayingTitle()
 
 bool CDVDPlayer::SwitchChannel(const CPVRChannelPtr &channel)
 {
+  const CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
+  if (channel && currentChannel && (*channel == *currentChannel))
```

Comment:
@xhaggi: AFAIK, <code>channel == currentchannel</code> would just compare the pointer values. I want to compare the actual channel objects.